tests/evaluation/test_evaluator.py

```diff
@@ -0,0 +1,398 @@
+    )
+
+    assert (gt_1.instance.numpy() == points_gt).all()
+    assert (pred_1.instance.numpy() == points_pred).all()
+
+    points = np.array(
+        [
+            [55.6, 30.2],
+            [10.1, 18.5],
+            [35.8, 12.0],
+        ]
+    )
+    assert (gt_3.instance.numpy() == points).all()
+
+
+def evaluator_case_2(user_labels, pred_labels):
+    # with no match frame pairs
+
+    with pytest.raises(Exception) as exc:
+        eval = Evaluator(user_labels, pred_labels)
+    assert str(exc.value) == "Empty Frame Pairs. No match found for the video frames"
+
+
+def evaluator_case_3(user_labels, pred_labels):
+    # with more predicted instances than user labeled instances
+    # one user lf with no match frame pair in predicted lf
+
+    eval = Evaluator(user_labels, pred_labels)
+    assert len(eval.frame_pairs) == 1
+    assert len(eval.positive_pairs) == 2
+    assert len(eval.false_negatives) == 0
+
+    eval = Evaluator(user_labels, pred_labels, match_threshold=1)
+    assert len(eval.frame_pairs) == 1
+    assert len(eval.positive_pairs) == 0
+    assert len(eval.false_negatives) == 2
+
+
+def evaluator_metrics(labels):
+    user_labels, pred_labels = labels["labels_1"]
+    eval = Evaluator(user_labels, pred_labels)
+
+    # test compute_instance_area
+    user_lf = user_labels[0]
+    points_gt = user_lf.numpy()[0]
+    area = compute_instance_area(points_gt)
+    area[0] == 77.14
+
+    # test compute distance function
+    dist_dict = eval.dists_dict
+    dists = dist_dict["dists"][0]
+    calc_dist = np.array([[4.0049968, 0.8, 1.3], [1.140175, 5.024937, np.nan]])
+    assert (np.abs(np.array(dists) - calc_dist[0]) <= 1e-5).all()
+
+    dists = np.array(dist_dict["dists"][1])
+    assert (np.abs(dists[:-1] - calc_dist[1][:-1]) <= 1e-5).all()
+    assert np.isnan(dists[-1])
+
+    # test visibility metrics
+    viz_metrics = eval.visibility_metrics()
+    assert viz_metrics["precision"] == float(1)
+    assert abs(viz_metrics["recall"] - float(0.833333)) <= 1e-5
+
+    # test distance_metrics
+    dist_metrics = eval.distance_metrics()
+    assert np.abs(dist_metrics["avg"] - 2.4540217) <= 1e-5
+    non_nans = np.array([4.0049968, 0.8, 1.3, 1.140175, 5.024937])
+    assert dist_metrics["p90"] - np.percentile(non_nans, 90) <= 1e-5
+
+    # test pck metrics
+    pck = eval.pck_metrics()
+    assert np.abs(pck["mPCK"] - 0.65) <= 1e-5
+
+    # test voc metrics
+    voc = eval.voc_metrics(match_score_by="pck")
+    assert np.abs(voc["pck_voc.recalls"][0] - 0.3333333) <= 1e-5
+    prec = np.zeros((101,))
+    prec[:34] = float(1) - np.spacing(1)
+    assert (voc["pck_voc.precisions"][0] == prec).all()
+
+    voc = eval.voc_metrics(match_score_by="oks")
+    assert np.abs(voc["oks_voc.recalls"][0] - 0.0) <= 1e-5
+
+    with pytest.raises(Exception) as exc:
+        eval.voc_metrics(match_score_by="moks")
+    assert (
+        str(exc.value)
+        == "Invalid Option for match_score_by. Choose either `oks` or `pck`"
+    )
```

Comment:
You can use the `match` keyword in [`pytest.raises`](https://docs.pytest.org/en/latest/reference/reference.html#pytest.raises) for this